app/javascript/driving/components/ActiveRide.js

```diff
@@ -1,124 +1,138 @@
 import React from 'react';
 import ContactVoter from './ContactVoter';
-import DispatchMatch from './DispatchMatch';
 
-const ActiveRide = props => {
-  const { ride, completeRide, pickupRider, cancelRide } = props;
-  let mapLink;
-  const passengers = 1 + parseInt(ride.additional_passengers, 10);
+const renderAddress = ride => (
+  <p>
+    {ride.status === 'picked_up' ? ride.to_address : ride.from_address}
+    <br />
+    {ride.status === 'picked_up'
+      ? `${ride.to_city}, ${ride.to_state}`
+      : `${ride.from_city}, ${ride.from_state}`}
+  </p>
+);
+
+const getRideLabel = status => {
+  if (status === 'driver_assigned') {
+    return 'Picking up';
+  }
+  return 'Dropping off';
+};
```

Comment:
There are lots of statuses that aren't `'driver_assigned'` or `'complete'` (which I'm guessing is the other option here). Even if there are only two possible statuses that could happen in this component at present, it would be more robust to external change if it were written as
```js
const getRideLabel = status => {
  if (status === 'driver_assigned') {
    return 'Picking up';
  }
  if (status === 'complete') {
    return 'Dropping off';
  }
  throw new Error(`Invalid status for ActiveRide: ${status}`)
};
```

Reply:
Great point, I think we need to map out all the possible ride statuses and what use cases we'd like to support in the app. Right now it more or less supports a "happy path" circa November 2016 😅